pints/_diagnostics.py

```diff
@@ -0,0 +1,60 @@
+#
+# Functions to calculate various MCMC diagnostics
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+import numpy as np
+
+
+def autocorrelation(lx):
+    """
+    Calculate autocorrelation for a vector x using a spectrum density
+    calculation.
+    """
+    xp = lx - np.mean(lx)
+    f = np.fft.fft(xp)
+    p = np.array([np.real(v)**2 + np.imag(v)**2 for v in f])
+    pi = np.fft.ifft(p)
+    return np.real(pi)[:lx.size / 2] / np.sum(xp**2)
+
+
+def autocorrelate_negative(l_autocorrelation):
```

Comment:
Consider using the naming scheme `_autocorrelate_negative` to mark these as private